### PR TITLE
docs: add documentation for viewer and editor apps

### PR DESCRIPTION
**Pull request by JammingBen**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/10533

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [x] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: docs/extension-system/viewer-editor-apps.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+geekdocEditPath: edit/master/docs/extension-system
+geekdocFilePath: viewer-editor-apps.md
+geekdocCollapseSection: true
+---
+
+{{< toc >}}
+
+## Viewer and editor apps
+
+The ownCloud Web allows developers to implement apps for viewing and editing specific file types. For instance, the built-in preview app serves as the default application for opening media files like images, videos, or audio.
```
Comment (dschmidt) [inline docs/extension-system/viewer-editor-apps.md:15]: "The ownCloud Web"  sounds weird

File: docs/extension-system/viewer-editor-apps.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+### Basic app structure
+
+An app is essentially a distinct package that must be specified as an external application in the Web configuration.
+
+The structure of an app is quite simple and straightforward. Consider, for example, the [draw.io app](https://github.com/owncloud/web/tree/master/packages/web-app-draw-io). It consists of a `package.json` file, a `src` directory containing all the source code, and a `l10n` directory for translations. Optionally, you may also include a `tests` directory if your application requires testing.
+
+### App setup
+
+Inside the `src` folder you will need a `index.ts` file that sets up the app so it can be registered by the Web runtime. It follows the basic structure as described in [the apps section]({{< ref "_index.md#apps" >}}), so it may look like this:
```
Comment (dschmidt) [inline docs/extension-system/viewer-editor-apps.md:27]:
```suggestion
Inside the `src` folder you will need an `index.ts` file that sets up the app so it can be registered by the Web runtime. It follows the basic structure as described in [the apps section]({{< ref "_index.md#apps" >}}), so it may look like this:
```

**Review by dschmidt — APPROVED** [review]
Love it! Are you going to extend this with details on the AppWrapper(Route)?

File: docs/extension-system/viewer-editor-apps.md (hunk 1 above)
Comment (JammingBen) [reply docs/extension-system/viewer-editor-apps.md:15]: I agree, just copied it over from the other sections. Simply "ownCloud Web" would sound better though IMO, I'll adjust it.

**Review by kulmann — COMMENTED** [review]
Could you add a link to this new docs page from within the `index.md` file in `docs/extension-system` please?

**Review by AlexAndBear — APPROVED** [review]

**Comment by JammingBen** [comment]
> Love it! Are you going to extend this with details on the AppWrapper(Route)?

Yeah we probably should (in a follow-up). Maybe you could assist me since you've built it and know best what it does and does not? 🤓

File: docs/extension-system/viewer-editor-apps.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  label?: string
+  name?: string
+  icon?: string
+  mimeType?: string
+  newFileMenu?: { menuTitle: () => string }
+  routeName?: string
+}
+```
+
+### Integration in Web
```
Comment (kulmann) [inline docs/extension-system/viewer-editor-apps.md:118]: For integration into web we now have what @fschade built. It's not available in oCIS 5.0.0, but will be in 5.1.0 / 6.0.0 (whatever comes next). A link to https://owncloud.dev/services/web/#web-apps should be sufficient instead of having this integration section.

The sentence at the end about developing from within the monorepo is a good addition though, at least until we have something better for dev setups of third party apps.

TL;DR: I'd remove this section again and place a link to the web service docs. But probably better in the `index.md` page since it applies for both third party extensions and apps. What do you think?

**Review by kulmann — APPROVED** [review]
🥳
